pyclient/pydeephaven/session.py

```diff
@@ -0,0 +1,296 @@
+                self._tables[t.name] = t.type
+
+        if response.removed:
+            for t in response.removed:
+                self._tables.pop(t.name, None)
+
+    # convenience/factory methods
+    def run_script(self, script: str) -> None:
+        """ Run the supplied Python script on the server.
+
+        Args:
+            script (str): the Python script code
+
+        Returns:
+
+        Raises:
+            DHError
+
+        """
+        with self._r_lock:
+            response = self.console_service.run_script(script)
+            self._parse_script_response(response)
+
+    def open_table(self, name: str) -> Table:
+        """ Open a table with the given name on the server.
+
+        Args:
+            name (str): the name of the table
+
+        Returns:
+            a Table object
+
+        Raises:
+            DHError
+
+        """
+        with self._r_lock:
+            return self.console_service.open_table(name)
+
+    def bind_table(self, table: Table, name: str) -> None:
+        """ Bind a table to the given name on the server so that it can be referenced by that name.
+
+        Args:
+            table (Table): a Table object
+            name (str): name for the table
+
+        Returns:
+
+        Raises:
+            DHError
+
+        """
+        with self._r_lock:
+            self.console_service.bind_table(table=table, variable_name=name)
+
+    def time_table(self, start_time: int = 0, period: int = 1000000000) -> TimeTable:
+        """ Create a time table on the server.
+
+        Args:
+            start_time (int): the start time for the time table in nano seconds, default is 0 (mid-night 1970.01.01)
+            period (int): the interval at which the time table ticks (adds a row), default is 1 second
+
+        Returns:
+            a TimeTable object
+
+        Raises:
+            DHError
+
+        """
+        return self.table_service.time_table(start_time=start_time, period=period)
+
+    def empty_table(self, size: int) -> EmptyTable:
+        """ create an empty table on the server.
+
+        Args:
+            size (int): the size of the empty table in number of rows
+
+        Returns:
+            a EmptyTable object
+
+        Raises:
+            DHError
+
+        """
+        return self.table_service.empty_table(size=size)
+
+    def import_table(self, data: pyarrow.Table) -> Table:
+        """ Import the pyarrow table as a new Deephaven table on the server.
+
+        Deephaven supports most of the Arrow data types. However, if the pyarrow table contains any field with a data
+        type not supported by Deephaven, the import operation will fail.
+
+        Args:
+            data (pyarrow.Table): a pyarrow Table object
+
+        Returns:
+            a Table object
+
+        Raises:
+            DHError
+
+        """
+        return self.flight_service.import_table(data=data)
+
+    def merge_tables(self, tables: List[Table], key_column: str = "") -> Table:
```

Comment:
I wonder if just `merge` is better?

Should we have `key_column` be `None` by default? Alternatively, I'm not sure we need to provide `key_column`/sorted functionality.